### ISSUE TITLE
[Bug] Not passing a value for `--repo-version` sets the package.json version to `true`

### ISSUE THREAD

**Issue by jeffdavidgreen** [lerna/lerna#229]
In `PublishCommand.js`, when setting the flag for `--repo-version`, if you don't set a value for the version, it changes all the updated packages version numbers to true.

The code I'm referring to is here:

``` javascript
if (this.flags.repoVersion) {
    return callback(null, {
        version: this.flags.repoVersion
    });
}
```

The command I ran is `lerna publish --repo-version`. I was thinking about this and I assumed that if I did not specify a version, then maybe the code could use the version found in each `package.json` file for the individual packages in the repo. For our workflow, we cannot have the prompts for the deployment scripts, so we were thinking about having developers increment the versions manually using `lerna publish --skip-npm` and then running on the build and deployment server `lerna bootstrap && lerna publish --skip-git --yes --repo-version`, but then we ran across this problem.

One solution I thought of is to include logic where if you don't add a version number as a value for `--repo-version`, then it will use the package.json version for each package.

The other solution is to include a new flag like `--package-version` that will do what I'm hoping to accomplish.

**Comment by davidfateh** [comment]
I am also seeing this bug.

**Comment by hzoo** [comment]
`getVersionsForUpdates` gets the next version to update to. If you're package is currently v1.0.0 and update it and do `lerna publish --repo-version` it doesn't know what version you want to update to. If you picked the same version you would get an error that it is trying to publish an existing version.

If you look at the other if/else branches in that code, it does 4 different checks otherwise (independent vs non-independent and canary vs non-canary). Another option that I want to add in would be using a github label/commit prefix to specify patch/minor/major so you wouldn't need a prompt.

**Comment by davidfateh** [comment]
A commit prefix might be helpful for when a user wants to update a version but doesn't want to use the prompt... Let's say the dev makes a PR, updates a package version and simply wants to use Lerna to then publish that package after the PR (with version number) has been approved.

**Comment by hzoo** [comment]
> so we were thinking about having developers increment the versions manually using lerna publish --skip-npm

I see, so you already updated the package.json version and want a way to use that, sounds ok to me

**Comment by davidfateh** [comment]
> I see, so you already updated the package.json version and want a way to use that, sounds ok to me


That seems to allow the dev to determine the individual package version while still utilizing lerna for its mass-publishing capabilities. This could help automate flows with build tools instead of having the build tools interface with the prompt (or some version of that).

**Comment by davidfateh** [comment]
I created a PR for `--package-version`. Please let me know if it makes sense for this project! I think it would help out greatly for what I would like to use Lerna for.

**Comment by thejameskyle** [comment]
Not passing a value to `--repo-version` should be an error.

**Comment by doug-wade** [comment]
This was fixed in #719
